pkgs/by-name/bi/bitwarden-desktop/package.nix

```diff
@@ -1,199 +1,20 @@
-    rustc
-    rustPlatform.cargoCheckHook
-    rustPlatform.cargoSetupHook
-  ];
-
-  buildInputs = [
-    glib
-    gtk3
-    libsecret
-  ];
-
-  preBuild = ''
-    if [[ $(jq --raw-output '.devDependencies.electron' < package.json | grep -E --only-matching '^[0-9]+') != ${lib.escapeShellArg (lib.versions.major electron.version)} ]]; then
-      echo 'ERROR: electron version mismatch'
-      exit 1
-    fi
-  '';
-
-  postBuild = ''
-    pushd apps/desktop
-
-    # desktop_native/index.js loads a file of that name regarldess of the libc being used
-    mv desktop_native/desktop_native.* desktop_native/desktop_native.linux-x64-musl.node
-
-    npm exec electron-builder -- \
-      --dir \
-      -c.electronDist=${electron}/libexec/electron \
-      -c.electronVersion=${electron.version}
-
-    popd
-  '';
+  extraArgs = removeAttrs args [ "callPackage" ];
 
-  doCheck = true;
-
-  nativeCheckInputs = [
-    dbus
-    (gnome.gnome-keyring.override { useWrappedDaemon = false; })
-  ];
-
-  checkFlags = [
-    "--skip=password::password::tests::test"
-  ];
-
-  checkPhase = ''
-    runHook preCheck
-
-    pushd ${cargoRoot}
-    export HOME=$(mktemp -d)
-    export -f cargoCheckHook runHook _eval _callImplicitHook
-    export cargoCheckType=release
-    dbus-run-session \
-      --config-file=${dbus}/share/dbus-1/session.conf \
-      -- bash -e -c cargoCheckHook
-    popd
-
-    runHook postCheck
-  '';
-
-  installPhase = ''
-    runHook preInstall
-
-    mkdir $out
-
-    pushd apps/desktop/dist/linux-unpacked
-    mkdir -p $out/opt/Bitwarden
-    cp -r locales resources{,.pak} $out/opt/Bitwarden
-    popd
-
-    makeWrapper '${electron}/bin/electron' "$out/bin/bitwarden" \
-      --add-flags $out/opt/Bitwarden/resources/app.asar \
-      --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations}}" \
-      --set-default ELECTRON_IS_DEV 0 \
-      --inherit-argv0
-
-    pushd apps/desktop/resources/icons
-    for icon in *.png; do
-      dir=$out/share/icons/hicolor/"''${icon%.png}"/apps
-      mkdir -p "$dir"
-      cp "$icon" "$dir"/${icon}.png
-    done
-    popd
-
-    runHook postInstall
-  '';
-
-  desktopItems = [
-    (makeDesktopItem {
-      name = "bitwarden";
-      exec = "bitwarden %U";
-      inherit icon;
-      comment = description;
-      desktopName = "Bitwarden";
-      categories = [ "Utility" ];
-    })
-  ];
+  pname = "bitwarden-desktop";
 
-  meta = {
-    changelog = "https://github.com/bitwarden/clients/releases/tag/${src.rev}";
-    inherit description;
+  meta = with lib; {
     homepage = "https://bitwarden.com";
-    license = lib.licenses.gpl3;
-    maintainers = with lib.maintainers; [ amarshall kiwi ];
+    description = "A secure and free password manager for all of your devices";
+    license = licenses.gpl3;
     platforms = [ "x86_64-linux" ];
     mainProgram = "bitwarden";
+    maintainers = with lib.maintainers; [ amarshall kiwi ];
```

Comment:
I've fixed the ordering and set `sourceProvenance`. I wasn't sure if it was appropriate to set myself as the maintainer for the Darwin version, but if it is I can do that.

Reply:
@amarshall I'm willing to maintain the Darwin version that build from source rather than binaryNativeCode.
@Hjdskes Maybe you can move sourceProvenance: binaryNativeCode to a new package `bitwarden-desktop-bin`?